src/iterate.test.ts

```diff
@@ -30,6 +30,18 @@ describe('IteratorWithOperators', () => {
             assert.equal(iterator.next().done, true)
         })
     })
+    describe('flatten', () => {
```

Comment:
The main point for myself was to make sure that the type inference matches appropriately. Otherwise, it's just a smoke test that improves coverage. Perhaps it would be best to define the tests in a way to use them in both calling contexts.

Reply:
Got it. Let's leave this out for now to not duplicate code. Type inference should be tested with https://github.com/Microsoft/dtslint but I don't have that set up yet.
In the future I would see this library use functions like RxJS instead of prototype methods so that would eliminate the coverage concerns.